src/main/kotlin/org/cascadebot/cascadebot/commands/moderation/TempSlowmodeCommand.kt

```diff
@@ -0,0 +1,94 @@
+package org.cascadebot.cascadebot.commands.moderation
+
+import net.dv8tion.jda.api.Permission
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.TextChannel
+import net.dv8tion.jda.api.exceptions.PermissionException
+import org.cascadebot.cascadebot.commandmeta.CommandContext
+import org.cascadebot.cascadebot.commandmeta.MainCommand
+import org.cascadebot.cascadebot.commandmeta.Module
+import org.cascadebot.cascadebot.data.managers.ScheduledActionManager
+import org.cascadebot.cascadebot.permissions.CascadePermission
+import org.cascadebot.cascadebot.scheduler.ActionType
+import org.cascadebot.cascadebot.scheduler.ScheduledAction
+import org.cascadebot.cascadebot.utils.DiscordUtils
+import org.cascadebot.cascadebot.utils.FormatUtils
+import org.cascadebot.cascadebot.utils.ParserUtils
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+
+class TempSlowmodeCommand : MainCommand() {
+
+    override fun onCommand(sender: Member, context: CommandContext) {
+        // <interval> <duration>  [channel]
+        if (context.args.size < 2) {
+            context.uiMessaging.replyUsage()
+            return
+        }
+
+        val longInterval = ParserUtils.parseTextTime(context.getArg(0), false) // In milliseconds
+        if (longInterval / 1000 > 21600) {  // 6 hour maximum time imposed by API
+            context.typedMessaging.replyWarning(context.i18n("commands.slowmode.time_exceeded"))
+            return
+        }
+
+        val longDuration = ParserUtils.parseTextTime(context.getArg(1), false)
+        if (longDuration <= 0) {
+            context.typedMessaging.replyDanger(context.i18n("responses.invalid_duration"))
+            return
+        }
+
+        var channel: TextChannel = context.channel
+        if (context.args.size == 3) {
+            val tempChannel = DiscordUtils.getTextChannel(context.guild, context.getArg(2))
+            if (tempChannel != null) {
+                channel = tempChannel
+            } else {
+                context.typedMessaging.replyDanger(context.i18n("responses.cannot_find_channel_matching", context.getArg(1)))
+                return
+            }
```

Comment:
Why are you creating a temp channel variable here? You can just store it in the channel object, and if channel is null return. temp channel is a extra variable that doesn't need to exist.

Reply:
That could actually work. This was they way I suggested, didn't think of that Ryan 👌

Reply:
I may be testing it wrong but it seems as if channel cannot be null, so binary's solution probably is the way to go about this